py/farm_ng/core/events_file_reader.py

```diff
@@ -1,83 +1,97 @@
 import importlib
+from importlib.resources import Package
 import sys
 from collections import defaultdict
 from pathlib import Path
-from typing import Any
-from typing import BinaryIO
-from typing import Dict
+from typing import Any, Tuple
+from typing import cast
+from typing import DefaultDict
+from typing import IO
 from typing import List
 from typing import Optional
+import logging
 
 from farm_ng.core import event_pb2
 from farm_ng.core.uri import Uri
+from farm_ng.core import uri_pb2
+
+
+def parse_protobuf_descriptor(desc) -> Tuple[str, str]:
+    # parse a proto descriptor and extract the message name and package.
+    # See: https://developers.google.com/protocol-buffers/docs/reference/python-generated#invocation
+    # NOTE: the descriptor comes in the shape of `farm_ng.core.proto.Timestamp`
+    desc_list = desc.split(".")
+    name = desc_list[-1]
+    package = desc_list[:-2] + [name.lower() + "_pb2"]
+    return name, ".".join(package)
```

Comment:
@ethanrublee this reverse engineer how protobuf compiles

Reply:
note i found this does not work for protobufs in general, like from google.  This makes an assumption about how the protobuf packages are organized, so only works for farm-ng protobufs, which is OK for now.

Reply:
@ethanrublee this definitely won't work after pulling this to the adk. E.g an `OakFrame` gets the following package name `ModuleNotFoundError: No module named 'farm_ng.oak.oakframe_pb2'

Reply:
Yes we will need to port that data to this new format.